src/Foundation/Commerce/client/dataModel.Generated.ts

```diff
@@ -14,326 +14,354 @@
+    tags: string[];
+    variants: ProductVariantModel[];
+}
+export interface ProductVariantModel {
+    adjustedPrice: number;
+    brand: string;
+    currencySymbol: string;
+    customerAverageRating: number;
+    description: string;
+    displayName: string;
+    imageUrls: string[];
+    listPrice: number;
+    productId: string;
+    variantId: string;
+    stockStatusName: string;
+    tags: string[];
+    properties: { [key: string]: string };
+}
+export interface ShippingInfo {
+    shippingMethods: ShippingMethod[];
+}
+export interface ShippingMethod {
+    description: string;
+    externalId: string;
+    lineIds: string[];
+    name: string;
+    partyId: string;
+    shippingPreferenceType: string;
+}
+export interface ShippingOption {
+    description: string;
+    name: string;
+    shippingOptionType: number;
+    shopName: string;
+}
+export interface SortOptionModel {
+    displayName: string;
+    isSelected: boolean;
+}
+export interface TotalPrice {
+    currencyCode: string;
+    currencySymbol: string;
+    total?: number;
+    subtotal?: number;
+    handlingTotal?: number;
+    shippingTotal?: number;
+    taxTotal: number;
+    totalSavings?: number;
+}
+export interface FederatedPaymentInfo {
+    cardToken: string;
+    partyId: string;
+    paymentMethodId: string;
+}
+export interface PaymentMethod {
+    description: string;
+    externalId: string;
+}
+export interface PaymentOption {
+    description: string;
+    name: string;
+    paymentOptionTypeName: string;
+}
+// tslint:disable-next-line: no-use-before-declare
+export interface DeliveryInfo extends BaseCheckoutInfo {
+    newPartyId: string;
+    shippingOptions: ShippingOption[];
+}
+export interface Subdivision {
+    code: string;
+    name: string;
+}
+export interface BaseCheckoutInfo {
+    userAddresses: Address[];
+}
+export interface Address {
+    address1: string;
+    address2: string;
+    city: string;
+    country: string;
+    countryCode: string;
+    email: string;
+    externalId: string;
+    firstName: string;
+    isPrimary: boolean;
+    lastName: string;
+    name: string;
+    partyId: string;
+    state: string;
+    zipPostalCode: string;
+}
+export interface BillingInfo {
+    paymentClientToken: string;
+    paymentMethods: PaymentMethod[];
+    paymentOptions: PaymentOption[];
+}
+export interface VoidResult {
+}
+export interface CommerceQueryFacet {
+    displayName: string;
+    foundValues: FacetValue[];
+    name: string;
+    values: any[];
+}
+export interface CommerceQuerySort {
+    displayName: string;
+    name: string;
+}
+export interface FacetValue {
+    aggregateCount: number;
+    name: string;
+}
+export interface KeyValuePair<TKey, TValue> {
+    key: TKey;
+    value: TValue;
+}
+export interface OrderHistoryResultModel {
+    currentPageNumber: number;
+    orders: CartModel[];
+}
+export interface UserLoginModel {
+    email: string;
+    password: string;
+}
+export interface Order extends Cart {
```

Comment:
Why does Order extend Cart? This is 2 different entities, in my opinion. 
Order could have a link to cart (maybe) but why extend?

Reply:
Sitecore entities `Order` extends `Cart`. Old structure of this project do this too. Besides many of `Cart` fields uses in frontend.